Sources/TuistLoader/SwiftPackageManager/SettingsMapper.swift

```diff
@@ -34,18 +34,43 @@ struct SettingsMapper {
 
     // swiftlint:disable:next function_body_length
     func settingsDictionaryForPlatform(_ platform: PackageInfo.Platform?) throws -> TuistGraph.SettingsDictionary {
+        let platformSettings = try settingsForPlatform(platform?.platformName)
+
+        return try map(
+            settings: platformSettings,
+            headerSearchPaths: headerSearchPaths,
+            defines: ["SWIFT_PACKAGE": "1"],
+            swiftDefines: "SWIFT_PACKAGE"
+        )
+    }
+
+    func settingsForPlatforms(_ platforms: [PackageInfo.Platform]) throws -> TuistGraph.SettingsDictionary {
+        var resolvedSettings = try settingsDictionaryForPlatform(nil)
```

Comment:
What does it mean to get a settings dictionary for a platform `nil`? I'd iterate on the implementation to make the code more readable.

Reply:
If it's nil, it doesn't take a platform in the account. It was here before, but sure, let me refactor it 👍